protocols/gossipsub/src/error.rs

```diff
@@ -146,17 +147,8 @@ impl From<std::io::Error> for PublishError {
     }
 }
 
-impl fmt::Display for GossipsubHandlerError {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{:?}", self)
-    }
-}
-
-impl std::error::Error for GossipsubHandlerError {
-    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
-        match self {
-            GossipsubHandlerError::Io(io) => Some(io),
-            _ => None,
-        }
+impl From<prost_codec::Error> for GossipsubHandlerError {
+    fn from(error: prost_codec::Error) -> GossipsubHandlerError {
+        GossipsubHandlerError::Codec(error)
     }
```

Comment:
You can drop this if you use `#[from]` instead of `#[source]`, sorry should have said that earlier!